frontend/src/pages/changeEmail.tsx

```diff
@@ -2,12 +2,23 @@ import type { NextPage } from "next";
 import { ResetComponent } from "../components/util/resetComponent";
 import Navbar from "../components/util/navBar";
 import { saveEmailOfUser } from "../api/calls/userCalls";
+import { emptyUser } from "../api/entities/UserEntity";
+import { useCurrentUser } from "../hooks/useCurrentUser";
 
 const ChangeEmail: NextPage = () => {
+    let { user: userResponse, error } = useCurrentUser();
+
+    const user = userResponse || emptyUser;
```

Comment:
You are blessed by the routeGuard. (I think) you can just ignore the underfined by using `user!`